jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -1366,8 +1368,10 @@ ecma_is_string_magic (const ecma_string_t *string_p, /**< ecma-string */
      * should return ecma-string with ECMA_STRING_CONTAINER_MAGIC_STRING
      * container type if new ecma-string's content is equal to one of magic strings.
      */
+#ifndef JERRY_NDEBUG
     JERRY_ASSERT (ecma_string_get_length (string_p) > LIT_MAGIC_STRING_LENGTH_LIMIT
                   || !ecma_is_string_magic_longpath (string_p, out_id_p));
+#endif /* !JERRY_NDEBUG */
```

Comment:
I would to keep the assertion in the current form, so it isn't a temporary change.